src/raft/raft.go

```diff
@@ -589,125 +631,150 @@ func (rf *Raft) tickerAsLeader() {
 					newNextI := min(reply.LastLogIndex+1, rf.nextIndexes[i]-1)
 					for ; newNextI > 1 && rf.log[newNextI].Term > reply.LastLogTerm; newNextI-- {
 					}
-					rf.nextIndexes[i] = max(1, newNextI) // speed up
+					rf.nextIndexes[i] = max(1, newNextI)                             // speed up
+					rf.nextIndexes[i] = max(rf.nextIndexes[i], rf.matchIndexes[i]+1) // IMPORTANT - SUBTLE BUG - as leader ticker is async, do not reduce nextIndexes if matchIndexes already updated! Otherwise, seeing nextIndexes jumps back and forth
```

Comment:
These places for speed-up follower log correction are important for passing `TestFigure8Unreliable2C`.
Here we speed up for the case where follower's `LastLogTerm` is less than leader's `rf.log[nextIndex].Term`, so we roll `nextIndex` to the smaller side until it reaches an try with the same term as follower's `LastLogTerm`.